tests/baselines/reference/arrayFilterBooleanExternalOverload4.types

```diff
@@ -49,37 +49,37 @@ const t11 = (0 as any as F2<1,1>)(id(),id());
 >id : ID
 
 const t12 = (0 as any as F2<1,2>)(id(),id());
->t12 : never
->(0 as any as F2<1,2>)(id(),id()) : never
+>t12 : 19
+>(0 as any as F2<1,2>)(id(),id()) : 19
 >(0 as any as F2<1,2>) : F2<1, 2>
 >0 as any as F2<1,2> : F2<1, 2>
 >0 as any : any
 >0 : 0
 >id() : (i: 1) => 1
 >id : ID
->id() : (i: 1) => 1
+>id() : (i: 2) => 2
 >id : ID
 
 const t21 = (0 as any as F2<2,1>)(id(),id());
->t21 : never
->(0 as any as F2<2,1>)(id(),id()) : never
+>t21 : 91
+>(0 as any as F2<2,1>)(id(),id()) : 91
 >(0 as any as F2<2,1>) : F2<2, 1>
 >0 as any as F2<2,1> : F2<2, 1>
 >0 as any : any
 >0 : 0
->id() : (i: 1) => 1
+>id() : (i: 2) => 2
 >id : ID
 >id() : (i: 1) => 1
 >id : ID
 
 const t22 = (0 as any as F2<2,2>)(id(),id());
->t22 : never
->(0 as any as F2<2,2>)(id(),id()) : never
+>t22 : 99
+>(0 as any as F2<2,2>)(id(),id()) : 99
 >(0 as any as F2<2,2>) : F2<2, 2>
 >0 as any as F2<2,2> : F2<2, 2>
 >0 as any : any
 >0 : 0
->id() : (i: 1) => 1
+>id() : (i: 2) => 2

```

Comment:
[Note that the incorrect `id()` value is not only cosmetic, it affects the final result.](https://github.com/microsoft/TypeScript/pull/56373/commits/2317daad176e8f99f2d41394a632434a0738330e#diff-8939d872f4bdf87cb549a3feb2fce124b57f681cd64b7536f2485b42654993b0)